src/lib/features/feature-toggle/configuration-revision-service.ts

```diff
@@ -51,14 +53,36 @@ export default class ConfigurationRevisionService extends EventEmitter {
         return ConfigurationRevisionService.instance;
     }
 
-    async getMaxRevisionId(): Promise<number> {
+    async getMaxRevisionId(environment?: string): Promise<number> {
+        if (environment && !this.maxRevisionId[environment]) {
+            await this.updateMaxEnvironmentRevisionId(environment);
+        }
+        if (
+            environment &&
+            this.maxRevisionId[environment] &&
+            this.maxRevisionId[environment] > 0
+        ) {
+            return this.maxRevisionId[environment];
+        }
         if (this.revisionId > 0) {
             return this.revisionId;
         } else {
             return this.updateMaxRevisionId();
         }
     }
 
+    async updateMaxEnvironmentRevisionId(environment: string): Promise<number> {
+        const envRevisionId = await this.eventStore.getMaxRevisionId(
+            this.maxRevisionId[environment],
+            environment,
+        );
+        if (this.maxRevisionId[environment] ?? 0 < envRevisionId) {
+            this.maxRevisionId[environment] = envRevisionId;
+        }
+
+        return this.maxRevisionId[environment];
+    }
+
```

Comment:
Same here. I also don't think you should directly access the map. You should use the `get` and `set` methods instead.

```ts
async updateMaxEnvironmentRevisionId(environment: string): Promise<number> {
    let maxRevId = this.maxRevisionId.get(environment) ?? 0;
    
    const envRevisionId = await this.eventStore.getMaxRevisionId(
        maxRevId,
        environment,
    );
    
    if (maxRevId < envRevisionId) {
        this.maxRevisionId.set(environment, envRevisionId);
        maxRevId = envRevisionId;
    }

    if (this.flagResolver.isEnabled('debugEtag')) {
        this.logger.info(
            `[etag] Computed ETag for environment ${environment}: ${envRevisionId} stored as "${maxRevId}"`,
        );
    }

    return maxRevId;
}
```